### PR TITLE
handle HTTP status 204 with content-encoding: gzip

### PR DESCRIPTION
**Pull request by sheltonial**
Fixes issue https://github.com/visionmedia/superagent/issues/857

### CODE REVIEW

**Comment by focusaurus** [comment]
LGTM. :+1:

**Comment by sheltonial** [comment]
Actually.. I think it needs a little more work. There's `pipe` function which also attempts to unzip.

File: lib/node/index.js [hunk 1]
```diff
@@ -430,7 +430,9 @@ Request.prototype.pipe = function(stream, options){
     }
 
     if (/^(deflate|gzip)$/.test(res.headers['content-encoding'])) {
-      res.pipe(zlib.createUnzip()).pipe(stream, options);
+      if (res.headers['content-length'] && res.headers['content-length'] > 0) {
+        res.pipe(zlib.createUnzip()).pipe(stream, options);
```
Comment (pornel) [inline lib/node/index.js]: what about `chunked` transfer encoding?

**Comment by pornel** [comment]
Thank you for the PR. I've merged it, but also allowed content-length header to be missing in order to support gzipped chunked encoding.

**Pull request merged**